rfc-022-makefile-standards.md

```diff
@@ -0,0 +1,90 @@
+# Makefile standards
+
+## Summary
+
+A set of standards for Makefiles across all projects, to supersede RFC012.
+
+## Problem
+
+When setting up/jumping onto a new project - sometimes it can be hard to discover what commands need to be run to perform common tasks (e.g. Running tests, serving the local development environment). This can make switching between projects more difficult as there can be a large number of ways to perform all of these tasks.
+
+Having a standard set of commands that can be run across all projects for common tasks allows people to get projects up and running without having to spend time looking into how things are set up to know what specific commands to run.
+
+## Proposal
+
+All projects **MUST** have a Makefile.
+
+All projects **MUST** have any relevant targets defined as [`PHONY` targets](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html) specified on just the first line of the file.
```

Comment:
propose this is changed to:

... specified in **ONE** of:
- the first line of the Makefile; **OR**
- the last line of the Makefile; **OR**
- defined above any rule that should be a `PHONY` target, and only that rule


----

this adds a little flexibility, and each variant has their merits.
The first 2 keep it central and out the way, however increases overhead of keeping rules + phony in sync,
the last one adds a bit of overhead when initially looking at a file, however has lower maintenance overhead as it keeps rules + PHONY together (easier to copy, easier to spot issues)

Reply:
I'd rather have one option agreed and have that as our standard rather than not coming to a consensus. Consistency across projects is a good thing IMO. Everyone just learns to do something one way, and it just works, and easier for everyone to pick up.

Although I'm leaning towards the last option to avoid things getting out of sync, historically I've found the need of even having `PHONY` targets to be minimal, as it's only ever really a problem if you have a target name that matches a directory or file within a project. There are some minimal speed gains to be had too, though, minimal. 

So I think if we went with the first option, to improve the readability, you'd find out very quickly if you had a `Makefile` target that needed a `PHONY` target if you had forgotten to add it to the list already.